Makefile

```diff
@@ -1,30 +1,77 @@
-.PHONY: all binary node_image test ut ut-circle st st-ssl clean run-etcd run-etcd-ssl create-dind help
-
-# These variables can be overridden by setting an environment variable.
-LOCAL_IP_ENV?=$(shell ip route get 8.8.8.8 | head -1 | cut -d' ' -f8)
-ST_TO_RUN?=tests/st/
-UT_TO_RUN?=tests/unit/
-# Can exclude the slower tests with "-a '!slow'"
-ST_OPTIONS?=
-HOST_CHECKOUT_DIR?=$(shell pwd)
-
-CALICOCTL_DIR=calicoctl
-CALICOCTL_FILE=$(CALICOCTL_DIR)/calicoctl.py $(wildcard $(CALICOCTL_DIR)/calico_ctl/*.py) calicoctl.spec
-
-NODE_CONTAINER_DIR=calico_node
-NODE_CONTAINER_FILES=$(shell find calico_node/ -type f ! -name '*.created')
-
-WHEEL_VERSION=0.0.0
-
+.PHONY: all binary calico/node test ut ut-circle st st-ssl clean run-etcd run-etcd-ssl help clean_calico_node
 default: help
 all: test                ## Run all the tests
 binary: dist/calicoctl   ## Create the calicoctl binary
-node_image: calico_node/.calico_node.created ## Create the calico/node image
+calico/node: calico_node/.calico_node.created ## Create the calico/node image
+node_image: calico/node
 test: st ut              ## Run all the tests
 ssl-certs: certs/.certificates.created ## Generate self-signed SSL certificates
 
+###############################################################################
+# calico/node build. Contains the following areas
+# - Populate the calico_node/filesystem
+# - Build the container itself
+###############################################################################
+NODE_CONTAINER_DIR=calico_node
+NODE_CONTAINER_FILES=$(shell find $(NODE_CONTAINER_DIR)/filesystem/{etc,sbin} -type f)
+NODE_CONTAINER_CREATED=$(NODE_CONTAINER_DIR)/.calico_node.created
+NODE_CONTAINER_BIN_DIR=$(NODE_CONTAINER_DIR)/filesystem/bin
+NODE_CONTAINER_BINARIES=startup allocate-ipip-addr calico-felix bird calico-bgp-daemon confd
+
+calico-node.tar: $(NODE_CONTAINER_CREATED)
+	docker save --output $@ calico/node:latest
+
+# Get docker2aci from https://github.com/appc/docker2aci
+calico-node-latest.aci: calico-node.tar
+	docker2aci $<
+
+$(NODE_CONTAINER_CREATED): $(NODE_CONTAINER_DIR)/Dockerfile  $(addprefix $(NODE_CONTAINER_BIN_DIR)/,$(NODE_CONTAINER_BINARIES))
```

Comment:
Yes, that's the whole point of a Makefile. If I do two builds in a row the second is instant. If I change a file, then make knows what to rebuild